### PR TITLE
fix: implement CREATE INDEX CONCURRENTLY support

### PR DESCRIPTION
**Pull request by tjgreen42**
## Summary

- Implement proper `ambulkdelete` callback to report all indexed CTIDs during CIC validation
- Add SQL test verifying CIC works correctly without duplicate entries
- Add shell script tests for concurrent table operations during CIC

## Background

CREATE INDEX CONCURRENTLY requires the `ambulkdelete` callback to report all existing TIDs so `validate_index` can determine which rows need to be inserted. Without this, all rows would be re-inserted causing duplicates.

The fix adds two helper functions in `vacuum.c`:
- `tp_bulkdelete_memtable_ctids`: iterates through document lengths dshash to collect memtable CTIDs
- `tp_bulkdelete_segment_ctids`: iterates through segment document maps to collect segment CTIDs

## Testing

- SQL test `test/sql/concurrent_build.sql`:
  - Basic CIC index creation
  - Verification that index is marked valid
  - Verification that no duplicate entries are produced
  - DROP INDEX CONCURRENTLY
  - Regular CREATE INDEX for comparison

- Shell test `test/scripts/cic.sh`:
  - Concurrent INSERTs during CIC (100k documents)
  - Concurrent UPDATEs during CIC
  - Concurrent DELETEs during CIC
  - Mixed concurrent operations during CIC
  - Multiple concurrent CIC operations
  - Configured with lower spill thresholds to trigger multiple memtable spills

### CODE REVIEW

File: test/expected/coverage.out [hunk 1]
```diff
@@ -1,7 +1,7 @@
 -- Additional coverage tests for untested code paths
 -- Tests debug functions, segment dump, and text<@>text operator
 CREATE EXTENSION IF NOT EXISTS pg_textsearch;
-NOTICE:  extension "pg_textsearch" already exists, skipping
```
Comment (SteveLauC) [inline test/expected/coverage.out:4]: `test/sql/compression.sql` doesn’t `DROP EXTENSION pg_textsearch` in the cleanup section, so seeing an “extension already exists” message is expected. I’m curious why CI still passes after this change, though. 🫨

File: test/sql/concurrent_build.sql [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+);
+
+INSERT INTO cic_basic (content) VALUES
+    ('the quick brown fox jumps over the lazy dog'),
+    ('postgresql is a powerful database system'),
+    ('full text search with bm25 ranking');
+
+-- Create index concurrently - should complete without error
+CREATE INDEX CONCURRENTLY cic_basic_idx ON cic_basic USING bm25(content)
+  WITH (text_config='english');
```
Comment (SteveLauC) [inline test/sql/concurrent_build.sql]: Nit: I think it’s better to specify all configuration options when creating the index rather than relying on defaults. That way, if a default value changes, we won’t need to update the `expected/xxx.out` file

```
NOTICE:  Using index options: k1=1.20, b=0.75
```

File: src/am/vacuum.c [hunk 3]
```diff
@@ -11,29 +11,148 @@
...
+	if (!doc_lengths_table)
+		return 0;
+
+	/* Iterate through all entries and call the callback for each CTID */
+	dshash_seq_init(&seq_status, doc_lengths_table, false);
+	while ((entry = dshash_seq_next(&seq_status)) != NULL)
+	{
+		if (ItemPointerIsValid(&entry->ctid))
+		{
+			/* Call the callback - returns true if tuple should be deleted */
```
Comment (SteveLauC) [inline src/am/vacuum.c]:
```suggestion
			/* Call the callback */
```

This comment does not seem necessary here as we don't do vacuum in this function
